src/NUnitFramework/framework/Attributes/ParallelizableAttribute.cs

```diff
@@ -51,6 +52,30 @@ public ParallelizableAttribute(ParallelScope scope) : base()
             Properties.Add(PropertyNames.ParallelScope, scope);
         }
 
+        /// <summary>
+        /// Overridden to check for invalid combinations of settings
+        /// </summary>
+        /// <param name="test"></param>
+        public override void ApplyToTest(Test test)
+        {
+            base.ApplyToTest(test);
+
+            if (test.RunState == RunState.NotRunnable)
+                return;
+
+            if (_scope.HasFlag(ParallelScope.Self) && _scope.HasFlag(ParallelScope.None))
+                MarkTestInvalid(test, "Test may not be both Parallel and NonParallel");
+
+            if (test is TestMethod && _scope.HasFlag(ParallelScope.ContextMask))
+                MarkTestInvalid(test, "ParallelScope of a test method may not specify Children or Fixtures");
+        }
+
+        private void MarkTestInvalid(Test test, string message)
```

Comment:
It's already used by two attributes at least and could be used by more. I'd prefer a `Test.MakeNotRunnable(string reason)` instance method. I'm not quite following where that violates the independence of attributes. Anyone who can modify RunState and Properties could call this method and at least it'd encourage consistency.